cmd/guacone/cmd/analyze.go

```diff
@@ -0,0 +1,121 @@
+  Intersection Analysis: Determine the intersection of two SBOMs, highlighting common software components shared between them.
+  Union Analysis: Combine two SBOMs to create a unified SBOM, merging software component lists while maintaining version integrity.`,
+  Example: `
+  Ingest the SBOMs to analyze:
+  $ guacone collect files guac-data-main/docs/spdx/syft-spdx-k8s.gcr.io-kube-apiserver.v1.24.4.json
+  $ guacone collect files guac-data-main/docs/spdx/spdx_vuln.json 
+  Difference
+  $ guacone analyze --diff --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  Union
+  $ guacone analyze --union --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  Intersection
+
+  $ guacone analyze --intersect --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  `,
+  Run: func(cmd *cobra.Command, args []string) {
+    ctx := logging.WithLogger(context.Background())
+    httpClient := http.Client{}
+    gqlclient := graphql.NewClient(viper.GetString("gql-addr"), &httpClient)
+
+
+    //get necessary flags
+    dot, _ := cmd.Flags().GetBool("dot")
+    intersect, _ := cmd.Flags().GetBool("intersect")
+    diff, _ := cmd.Flags().GetBool("diff")
+    union, _ := cmd.Flags().GetBool("union")
+    all, _ := cmd.Flags().GetBool("all")
+    maxprint, _ := cmd.Flags().GetInt("maxprint")
+
+
+    if diff && intersect && union || diff && intersect || diff && union || intersect && union {
+      fmt.Println("Must specify only one of --diff, --intersect, --union")
+      return
+    }
+
+    //create graphs
+    graphs := analysis.HasSBOMToGraph(cmd, ctx, gqlclient)
+
+    if diff {
+      analysisGraph, analysisList := analysis.HighlightAnalysis(graphs[0], graphs[1], 0)
+      analysis.GenerateAnalysisOutput(analysisGraph, analysisList, all, dot, maxprint, 0 ,gqlclient)
+    } else if intersect {
+      analysisGraph, analysisList := analysis.HighlightAnalysis(graphs[0], graphs[1], 1)
+      analysis.GenerateAnalysisOutput(analysisGraph, analysisList, all, dot, maxprint, 1,gqlclient)
+    } else if union {
+      analysisGraph, analysisList := analysis.HighlightAnalysis(graphs[0], graphs[1], 2)
+      analysis.GenerateAnalysisOutput(analysisGraph, analysisList, all, dot, maxprint, 2,gqlclient)
+    } else {
+      fmt.Println("Must specify one of --diff, --intersect, --union")
+      return
+    }
+  },
+}
+
+
+
+func init() {
+
+  rootCmd.PersistentFlags().Bool("intersect", false, "compute intesection of given sboms")
```

Comment:
Don't add flags to rootCmd, this will show up on all `guacone` cmds if so. Use this `analyzeCmd` you are creating.
Also, cli options are normally under `pkg/cli/store.go` but this command has so many it might make sense to just leave them here.